test/harry/main/org/apache/cassandra/harry/model/ASTSingleTableModel.java

```diff
@@ -341,61 +498,80 @@ else if (!clusterings.isEmpty())
         }
     }
 
+    private static void maybeUpdateColumns(Set<Symbol> columns,
+                                           @Nullable BytesPartitionState.Row row,
+                                           long nowTs, Map<Symbol, Expression> set,
+                                           ColumnUpdate update)
+    {
+        if (columns.isEmpty())
+        {
+            update.update(nowTs, Collections.emptyMap());
+            return;
+        }
+        // static columns to add in.  If we are doing something like += to a row that doesn't exist, we still update statics...
+        Map<Symbol, ByteBuffer> write = new HashMap<>();
+        for (Symbol col : columns)
+        {
+            ByteBuffer current = row == null ? null : row.get(col);
+            EvalResult result = eval(col, current, set.get(col));
+            if (result.kind == EvalResult.Kind.SKIP) continue;
+            write.put(col, result.value);
+        }
+        if (!write.isEmpty())
+            update.update(nowTs, write);
+    }
+
     public boolean shouldApply(Mutation mutation)
     {
         if (!mutation.isCas()) return true;
         return shouldApply(mutation, selectPartitionForCAS(mutation));
     }
 
-    private SelectResult selectPartitionForCAS(Mutation mutation)
+    private CasContext selectPartitionForCAS(Mutation mutation)
     {
-        var partition = partitions.get(factory.createRef(pd(mutation)));
-        if (partition == null) return SelectResult.ordered(factory.selectionOrder, NO_ROWS);
-
-        var cd = cdOrNull(mutation);
-        var row = cd == null ? null : partition.get(cd);
-        ImmutableUniqueList<Symbol> columns = cd != null ? factory.selectionOrder : factory.partitionAndStaticColumns;
-        return SelectResult.ordered(columns, new ByteBuffer[][] { getRowAsByteBuffer(columns, partition, row)});
+        BytesPartitionState.Ref ref = referencePartition(mutation);
+        Clustering<ByteBuffer> cd = cdOrNull(mutation);
+        BytesPartitionState partition = partitions.get(ref);
+        return new CasContext(ref, cd, partition);
     }
 
-    private boolean shouldApply(Mutation mutation, SelectResult current)
+    private boolean shouldApply(Mutation mutation, CasContext ctx)
```

Comment:
the reason for this change is that CAS's apply for partition level queries is based off the static row and not the partition existing!  `SelectResult` mimics what a user would see, which is not how CAS works.  

In order to validate CAS apply we need access to the row and static row separately